platform/ui/src/components/InputFilterText/InputFilterText.tsx

```diff
@@ -0,0 +1,91 @@
+import classNames from 'classnames';
+import debounce from 'lodash.debounce';
+import React, {
+  ReactElement,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+
+import Icon from '../Icon';
+
+type InputFilterTextProps = {
+  className?: string;
+  value?: string;
+  placeholder: string;
+  onDebounceChange?: (val: string) => void;
+  onChange?: (val: string) => void;
+  debounceTime?: number;
+};
+
+/**
+ * A component to use as the input for text to filter by/on. A debounced callback is automatically provided
+ * so that the filtering in-turn will be debounced. There is also a straight onChange callback when the filter value is
+ * required immediately and NOT debounced. The debounce time is also configurable.
+ */
+const InputFilterText = ({
```

Comment:
can we have a storybook for this? follow the rest of the stories to how to add it. 
You can run the storybook with 

```
cd platform/ui
yarn start or dev (i don't remember)
```